### PR TITLE
Scripts for feature "Handling vr help requests"

### PR DESCRIPTION
**Pull request by GetmanetsIrina**
ATF Test Scripts to check #[2154](https://github.com/smartdevicelink/sdl_core/issues/2154)

This PR is **ready** for review.

### Summary
Scripts for feature "Handling vr help requests"

### ATF version
[Develop branch](https://github.com/smartdevicelink/sdl_atf/tree/develop)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by askirk** [comment]
@theresalech ready for review

**Comment by jacobkeeler** [comment]
@GetmanetsIrina Tests need to be updated with the inclusion of [this revision](https://github.com/smartdevicelink/sdl_evolution/pull/519) which removes the use of the 10 second timer

**Comment by GetmanetsIrina** [comment]
@jacobkeeler , thanks for information, updating is in progress.

**Comment by dboltovskyi** [comment]
@jacobkeeler Updating of test scripts on a last stage (review) and is going to be finished tomorrow.

**Comment by dboltovskyi** [comment]
@jacobkeeler Test scripts have been updated according to the last changes in sdl_core. Please check.

File: test_scripts/Handling_VR_help_requests/002_SetGlobalProp_commands_with_and_without_vrCommand.lua [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local commandWithoutVr = {
+  cmdID = 2,
+  menuParams = {
+    menuName = "CommandWithoutVr"
+  }
+}
+local commandWithtVr = {
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/Handling_VR_help_requests/002_SetGlobalProp_commands_with_and_without_vrCommand.lua]: @jacobkeeler Please check 0cef533

File: test_scripts/Handling_VR_help_requests/005_SetGlobalProp_apply_limitation_only_for_addcommand.lua [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+runner.Step("App activation", common.activateApp)
+
+runner.Title("Test")
+for i = 1, 30 do
+  runner.Step("SetGlobalProperties from SDL after added command" ..i, common.addCommandWithSetGP, { i })
+end
+for i = 1, 10 do
+  runner.Step("SetGlobalProperties from SDL after deleted command" ..i, common.deleteCommandWithSetGP, { i })
+end
+runner.Step("Absence SetGlobalProperties from SDL after adding 34 command", common.addCommandWithoutSetGP, { 31 })
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/Handling_VR_help_requests/005_SetGlobalProp_apply_limitation_only_for_addcommand.lua]: @jacobkeeler Please check 0cef533

File: test_scripts/Handling_VR_help_requests/002_SetGlobalProp_commands_with_and_without_vrCommand.lua [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  cmdID = 2,
+  menuParams = {
+    menuName = "CommandWithoutVr"
+  }
+}
+local commandWithVr = {
+  cmdID = 1,
+  vrCommands = { "vrCommand"},
+  menuParams = {
+    menuName = "commandWithtVr"
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/Handling_VR_help_requests/002_SetGlobalProp_commands_with_and_without_vrCommand.lua]: @jacobkeeler Please check 1ebce4c700c51a08a6d6c8a2009afcf91575e088

File: test_scripts/Handling_VR_help_requests/016_SetGlobalProp_commands_with_and_without_vrCommand_by_resumption.lua [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local commandWithoutVr = {
+  cmdID = 2,
+  menuParams = {
+    menuName = "CommandWithoutVr"
+  }
+}
+local commandWithtVr = {
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/Handling_VR_help_requests/016_SetGlobalProp_commands_with_and_without_vrCommand_by_resumption.lua]: @jacobkeeler Please check 1ebce4c700c51a08a6d6c8a2009afcf91575e088

File: test_scripts/Handling_VR_help_requests/021_SetGlobalProp_absence_after_custom_request_before_resumption.lua [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+          "Actual result:" .. common.tableToString(data.params.vrCommands) .. "\n" ..
+          "Expected result:" .. common.tableToString(value.vrCommand) .."\n"
+        end
+        return vrCommandCompareResult, Msg
+      end
+    end
+    return true
+  end)
+  :Times(#common.commandArray)
+EXPECT_HMICALL("UI.AddCommand")
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/Handling_VR_help_requests/021_SetGlobalProp_absence_after_custom_request_before_resumption.lua]: @jacobkeeler Please check 1ebce4c700c51a08a6d6c8a2009afcf91575e088

File: test_scripts/Handling_VR_help_requests/029_SetGlobalProp_only_successful_added_commands.lua [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  :Times(0)
+
+  EXPECT_HMICALL("TTS.SetGlobalProperties")
+  :Times(0)
+
+  EXPECT_HMICALL("UI.AddCommand")
+  :Do(function(_,data)
+    hmiConnection:SendError(data.id, data.method, "REJECTED", "Rejected request")
+  end)
+  local requestUiParams = {
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/Handling_VR_help_requests/029_SetGlobalProp_only_successful_added_commands.lua]: @jacobkeeler Please check 1ebce4c700c51a08a6d6c8a2009afcf91575e088

File: test_scripts/Handling_VR_help_requests/commonVRhelp.lua [hunk 7]
```diff
@@ -0,0 +1,399 @@
...
+  }
+  return allParams
+end
+
+function m.getAddCommandParams(pN)
+  local out = {
+    cmdID = pN,
+    vrCommands = { "vrCommand" .. pN},
+    menuParams = {
+    menuName = "Command" .. pN
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/Handling_VR_help_requests/commonVRhelp.lua]: @jacobkeeler Please check 1ebce4c700c51a08a6d6c8a2009afcf91575e088

File: test_scripts/Handling_VR_help_requests/commonVRhelp.lua [hunk 8]
```diff
@@ -0,0 +1,399 @@
...
+  local mobSession = m.getMobileSession(pAppId)
+  local hmiConnection = m.getHMIConnection()
+  local cid = mobSession:SendRPC("AddCommand", pParams)
+  EXPECT_HMICALL("UI.AddCommand")
+  :Do(function(_,data)
+    hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+  end)
+  if pParams.vrCommands then
+    table.insert(m.commandArray, { cmdID = pParams.cmdID, vrCommand = pParams.vrCommands})
+    local requestUiParams = {
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/Handling_VR_help_requests/commonVRhelp.lua]: @jacobkeeler Please check 1ebce4c700c51a08a6d6c8a2009afcf91575e088

File: test_scripts/Handling_VR_help_requests/commonVRhelp.lua [hunk 9]
```diff
@@ -0,0 +1,399 @@
...
+  if pParams.requestParams.helpPrompt or pParams.requestParams.timeoutPrompt then
+    EXPECT_HMICALL("TTS.SetGlobalProperties", pParams.requestTtsParams)
+    :Do(function(_,data)
+      hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  end
+  mobSession:ExpectResponse(cid, { success = true, resultCode = "SUCCESS"})
+end
+
+function m.setGlobalPropertiesFromSDL(pIsCheckOnTimeoutRequied)
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/Handling_VR_help_requests/commonVRhelp.lua]: @jacobkeeler Please check 1ebce4c700c51a08a6d6c8a2009afcf91575e088

File: test_scripts/Handling_VR_help_requests/commonVRhelp.lua [hunk 10]
```diff
@@ -0,0 +1,399 @@
...
+end
+
+function m.activateApp(pAppId)
+  if not pAppId then pAppId = 1 end
+  local requestId = m.getHMIConnection():SendRequest("SDL.ActivateApp", { appID = m.getHMIAppId(pAppId) })
+  m.getHMIConnection():ExpectResponse(requestId)
+  m.getMobileSession(pAppId):ExpectNotification("OnHMIStatus",
+    { hmiLevel = "FULL", audioStreamingState = "AUDIBLE", systemContext = "MAIN" })
+    :Do(function()
+      m.timeActivation = timestamp()
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/Handling_VR_help_requests/commonVRhelp.lua]: @jacobkeeler Please check 1ebce4c700c51a08a6d6c8a2009afcf91575e088

File: test_scripts/Handling_VR_help_requests/commonVRhelp.lua [hunk 11]
```diff
@@ -0,0 +1,399 @@
...
+function m.reconnect(pAppId)
+  if not pAppId then pAppId = 1 end
+  m.getMobileSession(pAppId):Stop()
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered",
+    {appID = m.getHMIAppId(pAppId), unexpectedDisconnect = true})
+  :Do(function()
+    test.mobileSession[pAppId] = mobile_session.MobileSession(
+    test,
+    test.mobileConnection,
+    config["application" .. pAppId].registerAppInterfaceParams)
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/Handling_VR_help_requests/commonVRhelp.lua]: @jacobkeeler Please check 1ebce4c700c51a08a6d6c8a2009afcf91575e088

**Pull request merged**